prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -44,9 +47,12 @@ def run(
         )
         agent.run(self.n_iterations)
         long_term_memory.save_history(agent.history)
-        prompt_handler.save_prompt(agent.system_prompt)
+        editable_prompt = get_editable_prompt_from_agent(agent)
```

Comment:
Otherwise fails with `The system prompt already contains the non-updatable part.` inside `build_full_system_prompt` when loading a `historical_prompt`